### DISCUSSION TITLE
PUBLIC_URL not working properly

### DISCUSSION THREAD

**Discussion by btzdnl** [searx/searx#2443] [Installation issue: Docker]
I just tried upgrading from searx 0.16, however all more recent versions use incorrect URLs…?

I start it like this:

```bash
docker run --detach --restart always -p 127.0.0.2:8080:8080 -e BASE_URL=http://192.168.2.5/searx --name searx searx/searx
```

It is reverse-proxied using nginx like this:

```
        location /searx {
            proxy_pass http://127.0.0.2:8080;
            proxy_set_header        Host                 $host;
            proxy_set_header        X-Real-IP            $remote_addr;
            proxy_set_header        X-Forwarded-For      $proxy_add_x_forwarded_for;
            proxy_set_header        X-Remote-Port        $remote_port;
            proxy_set_header        X-Forwarded-Proto    $scheme;
            proxy_set_header        X-Script-Name        /searx;
            proxy_redirect          off;
        }
```

However, all the URLs now have a duplicate `/searx` in them, like this: `http://192.168.2.5/searx/searx/translations.js`

What’s the deal here? I don’t think my configuration is all that special.

---

Further investigation has revealed that using the wrong `BASE_URL` allows search to work, but results in an incorrect URL in the OpenSearch XML, like this: `http://192.168.2.5/autocompleter?q={searchTerms}`

**Comment by unixfox** [comment]
What if you try with just `BASE_URL=http://192.168.2.5` without the /searx?

**Reply by btzdnl** [reply]
That works. By accident. The URL appears to be constructed like this:

    $BASE_URL/$X_SCRIPT_NAME/$ACTUAL_PATH

If I remove the `X-Script-Name` header the URLs are correct but searx’ routing does not work. So maybe I’m doing something wrong or the documentation is somehow wrong/unclear?

**Reply by unixfox** [reply]
I've to summon @return42 on that, I don't really have an answer for your question.
